### DISCUSSION TITLE
Routing Guitar Pro or YouTube audio

### DISCUSSION THREAD

**Discussion by wt0m** [jamulussoftware/jamulus#1996] [General]
Hi everyone!

I'm wondering if it is possible to mix the sound from an audio interface and the sound of a Windows application while talking. In other words, I will have 3 audio sources: my guitar (through the audio interface), an app (like Guitar Pro, or YouTube, or Windows Media Player) and my laptop's microphone. The app would allow me to play backing tracks and I would like everyone in the session to hear them.

Also, I'm new to the world of recording and I don't have an audio interface yet, but I'm hesitating between the following : Motu M2/M4, Focusrite Scarlett Solo/2i2/4i4, SSL 2, Steinberg UR24C. Will they all work with Jamulus? Which one would you recommend for recording rock/heavy metal/neoclassical music?

Any help would be greatly appreciated, thanks 😃

**Comment by gilgongo** [comment]
I've not tried it myself, but the reccommended way of having more than one thing outputting to a server is to use multiple clients, each using a different channel in your audio device I think. Personally, I would simply plug a mic and my guitar into an external mixing desk though, and just have one client on my machine for simplicity sake. Apps are a bit more tricky, since you can route the output of these via software (eg Voicemeeter or Jack) but there are latency implications to this. So again, I'd just have a laptop running their outputs straight out to a mixing desk into which the guitar and mic were also plugged in.

But I'm sure others would do it differently.

As to audio interfaces, you might have a look over here for some of those: https://jamulus.io/kb/2021/01/05/Jamulus-Sound-Devices.html

**Comment by wt0m** [comment]
Thanks for the quick answer @gilgongo and for your advice!

I didn't think of using multiple clients on the same computer, great idea! However, wouldn't this affect the latency? I guess it will at least use much more bandwidth as each client sends and receives data from the server.
I checked Voicemeeter and Jack, they seem to be the solution for app output routing, do you have an idea about how much latency they can add?

Unfortunately, I don't have a mic so I guess I'll have to use Voicemeeter/Jack. If I understand the idea, the following setup could work, right?
`client 1 <- guitar via audio interface`
`client 2 <- Voicemeeter/Jack <- laptop mic + YT or Guitar Pro sound`

Also, thanks for the link with the audio interfaces, I ordered a Motu M4 as it seems to be one of the best in its price range.

**Reply by gilgongo** [reply]
Multiple clients won't hit latency unless you have an unusually underpowered machine. Similarly with bandwidth really unless you have a 1Mb/s line or something. 

Voicemeteer will add some latency, possibly as much as 20-30ms. Don't know about Jack though.

But yes, that config should work I think. But like I say, I've not tried it myself :-)

**Reply by wt0m** [reply]
Thanks again for your help @gilgongo !
I should receive my Motu in about 2 weeks as it is currently not in stock, I'll tell you if I managed to make everything work 😀
